core/src/main/java/io/grpc/internal/LogExceptionRunnable.java

```diff
@@ -56,7 +58,8 @@ public void run() {
       task.run();
     } catch (Throwable t) {
       log.log(Level.SEVERE, "Exception while executing runnable " + task, t);
-      throw t instanceof RuntimeException ? (RuntimeException) t : new RuntimeException(t);
+      Throwables.propagateIfPossible(t);
+      throw new AssertionError(t);
```

Comment:
The only classes that are [unchecked is RuntimeException and Error](https://docs.oracle.com/javase/specs/jls/se7/html/jls-11.html#jls-11.1.1) (and children). So getting to this line should not be possible (without hacks). So `AssertionError` seems fitting.